lib/extensions/rack_session_dalli_patch.rb

```diff
@@ -12,15 +12,24 @@ def delete_sessions(session_ids)
   # cause a load to occur. Consequently, we need to manage things
   # carefully to prevent a deadlock between the Rails Interlock and
   # Dalli's own exclusive lock.
-  def with_lock(*args)
+  def with_block(*args)
     ActiveSupport::Dependencies.interlock.permit_concurrent_loads do
-      super(*args) do
+      super(*args) do |dc|
         ActiveSupport::Dependencies.interlock.running do
-          yield
+          yield dc
         end
       end
     end
   end
 end
 
+# In commit: https://github.com/petergoldstein/dalli/commit/9f9c508afab263a2451f2209c4396daf98d33a1b
+# with_lock was renamed to with_block with a slightly different interface.
+# All versions since 2.7.7 have with_lock now.
+# Additionally, we'll detect and warn if the method we're about to prepend/override doesn't exist before we try to prepend it.
+begin
+  Rack::Session::Dalli.instance_method(:with_block)
+rescue NameError => err
+  warn "Dalli is missing the method we're about to prepend: #{err}.  Was the method removed or renamed upstream? See: #{__FILE__}"
```

Comment:
Could this be two `warn` lines?  My pedantic `80` char self is annoyed by how long this line is...

Reply:
This is not worth holding up the PR.  If you disagree, that is totally fine/valid.

Reply:
Get out of my head.  I had that originally and made it one line when it was a bit shorter and didn't go back to two lines...  👍

Reply:
`\(._.)> ~ ~ ~ [¬°-°]¬`